osf_tests/test_gv_utils.py

```diff
@@ -0,0 +1,312 @@
+
+    @pytest.fixture
+    def noncontributor(self):
+        return factories.AuthUserFactory()
+
+    @pytest.fixture
+    def resource(self, contributor):
+        return factories.ProjectFactory(creator=contributor)
+
+    @pytest.fixture
+    def external_service(self, mock_gv):
+        return mock_gv.configure_mock_provider('blarg')
+
+    @pytest.fixture
+    def external_account(self, mock_gv, contributor, external_service):
+        return mock_gv.configure_mock_account(contributor, external_service.name)
+
+    @pytest.fixture
+    def configured_addon(self, mock_gv, resource, external_account):
+        return mock_gv.configure_mock_addon(resource, external_account)
+
+    def test_validate_headers__bad_key(self, mock_gv, contributor, external_account):
+        request_url = f'{GRAVYVALET_URL}/v1/user-references/{external_account.account_owner_pk}/'
+        auth_headers = gv_auth.make_gravy_valet_hmac_headers(
+            request_url=request_url,
+            request_method='GET',
+            requesting_user=contributor,
+            hmac_key='bad key'
+        )
+        with mock_gv.run_mock():
+            resp = requests.get(request_url, headers=auth_headers)
+        assert resp.status_code == 403
+
+    def test_validate_headers__missing_headers(self, mock_gv, contributor, external_account):
+        request_url = f'{GRAVYVALET_URL}/v1/user-references/{external_account.account_owner_pk}/'
+        with mock_gv.run_mock():
+            resp = requests.get(request_url)
+        assert resp.status_code == 401
+
+    @pytest.mark.parametrize('subpath', ['', 'authorized_storage_accounts/'])
+    def test_validate_user__success(self, mock_gv, contributor, external_account, subpath):
+        request_url = f'{GRAVYVALET_URL}/v1/user-references/{external_account.account_owner_pk}/{subpath}'
+        auth_headers = gv_auth.make_gravy_valet_hmac_headers(
+            request_url=request_url,
+            request_method='GET',
+            requesting_user=contributor
+        )
+        with mock_gv.run_mock():
+            resp = requests.get(request_url, headers=auth_headers)
+        assert resp.status_code == 200
+
+    @pytest.mark.parametrize('subpath', ['', 'authorized_storage_accounts/'])
+    def test_validate_user__wrong_user(self, mock_gv, noncontributor, external_account, subpath):
+        request_url = f'{GRAVYVALET_URL}/v1/user-references/{external_account.account_owner_pk}/{subpath}'
+        auth_headers = gv_auth.make_gravy_valet_hmac_headers(
+            request_url=request_url,
+            request_method='GET',
+            requesting_user=noncontributor,
+        )
+        with mock_gv.run_mock():
+            resp = requests.get(request_url, headers=auth_headers)
+        assert resp.status_code == 403
+
+    @pytest.mark.parametrize('subpath', ['', 'authorized_storage_accounts/'])
+    def test_validate_user__no_user(self, mock_gv, noncontributor, external_account, subpath):
```

Comment:
nitpick: unused fixture
```suggestion
    def test_validate_user__no_user(self, mock_gv, external_account, subpath):
```

Reply:
Got a couple of other instances of this, too. Thanks!